mailbox/backup/src/test/java/org/apache/james/mailbox/backup/FlagExtraFieldTest.java

```diff
@@ -0,0 +1,310 @@
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getLocalFileDataLength().getValue())
+                .isEqualTo(17);
+        }
+    }
+
+    @Nested
+    class GetCentralDirectoryLength {
+
+        @Test
+        void getCentralDirectoryLengthShouldThrowWhenNoValue() {
+            FlagsExtraField testee = new FlagsExtraField();
+
+            assertThatThrownBy(() -> testee.getCentralDirectoryLength().getValue())
+                .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void getCentralDirectoryLengthShouldReturnIntegerSize() {
+            FlagsExtraField testee = new FlagsExtraField(new Flags());
+
+            assertThat(testee.getCentralDirectoryLength().getValue())
+                .isEqualTo(0);
+        }
+
+        @Test
+        void getCentralDirectoryLengthShouldReturnIntegerSizeWhenSystemFlagSet() {
+            Flags flags = new Flags();
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryLength().getValue())
+                .isEqualTo(5);
+        }
+
+        @Test
+        void getCentralDirectoryLengthShouldReturnIntegerSizeWhenUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            flags.add("newFlag");
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryLength().getValue())
+                .isEqualTo(15);
+        }
+
+        @Test
+        void getLocalFileDataLengthShouldReturnIntegerSizeWhenSystemAndUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            flags.add(Flags.Flag.ANSWERED);
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryLength().getValue())
+                .isEqualTo(23);
+        }
+    }
+
+    @Nested
+    class GetLocalFileData {
+
+        @Test
+        void getLocalFileDataDataShouldThrowWhenNoValue() {
+            FlagsExtraField testee = new FlagsExtraField();
+
+            assertThatThrownBy(() -> testee.getLocalFileDataData())
+                .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void getLocalFileDataDataShouldReturnByteArraysOfSystemFlagSet() {
+            Flags flags = new Flags();
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+            byte[] expectedData = new byte[]{92, 83, 69, 69, 78};
+
+            assertThat(testee.getLocalFileDataData())
+                .isEqualTo(expectedData);
+        }
+
+        @Test
+        void getLocalFileDataDataShouldReturnByteArrayOfUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            FlagsExtraField testee = new FlagsExtraField(flags);
+            byte[] expectedData = new byte[]{109, 121, 70, 108, 97, 103, 115};
```

Comment:
I am very bad at reviewing binary...

I must admit, if possible that reviewing the String representation of these byte fields will feel easier!